src/main.c

```diff
@@ -1,7 +1,172 @@
+            char response[BUFFER_SIZE];
+            snprintf(
+                    response, sizeof(response),
+                    "HTTP/1.0 404 Not Found\r\n"
+                    "Content-Type: text/html\r\n\r\n"
+                    "<h1>404 - Arquivo nao encontrado</h1>"
+            );
+            write(client_fd, response, strlen(response));
+            close(client_fd);
+            continue;
+        }
+
+        struct stat file_stat;
+        if (stat(caminho, &file_stat) == -1) { // verificar o caminho é válido
+            char response[BUFFER_SIZE];
+            snprintf(
+                    response, sizeof(response),
+                    "HTTP/1.0 404 Not Found\r\n"
+                    "Content-Type: text/html\r\n\r\n"
+                    "<h1>404 - Arquivo nao encontrado</h1>"
+            );
+            write(client_fd, response, strlen(response));
+            close(client_fd);
+            continue;
+        }
+
+        if (S_ISDIR(file_stat.st_mode)) { // Se for diretório, adiciona index.html
+            strncat(caminho, "/index.html", sizeof(caminho) - strlen(caminho) - 1);
+            if (stat(caminho, &file_stat) ==
+                -1) { // verifica se index.html existe naquele diretório
+                char response[BUFFER_SIZE];
+                snprintf(
+                        response, sizeof(response),
+                        "HTTP/1.0 404 Not Found\r\n"
+                        "Content-Type: text/html\r\n\r\n"
+                        "<h1>404 - Arquivo nao encontrado</h1>"
+                );
+                write(client_fd, response, strlen(response));
+                close(client_fd);
+                continue;
+            }
+        }
+
+        FILE *fp = fopen(caminho, "rb"); // executa o arquivo em binário
+        if (!fp) {
+            char response[BUFFER_SIZE];
+            snprintf(
+                    response, sizeof(response),
+                    "HTTP/1.0 500 Internal Server Error\r\n"
+                    "Content-Type: text/html\r\n\r\n"
+                    "<h1>500 - Erro ao abrir o arquivo</h1>"
+            );
+            write(client_fd, response, strlen(response));
+            close(client_fd);
+            continue;
+        }
+
+        char conteudo[BUFFER_SIZE];
+        int lidos = fread(conteudo, 1, sizeof(conteudo) - 1, fp);
+        conteudo[lidos] = '\0';
+        fclose(fp);
+
+        const char *content_type = "text/plain";
+        if (strstr(caminho, ".html"))
+            content_type = "text/html";
+        else if (strstr(caminho, ".css"))
+            content_type = "text/css";
+        else if (strstr(caminho, ".js"))
+            content_type = "application/javascript";
+
+        char response[BUFFER_SIZE];
+        snprintf(
+                response, sizeof(response),
+                "HTTP/1.0 200 OK\r\n"
+                "Content-Type: %s\r\n"
+                "Content-Length: %d\r\n\r\n",
+                content_type, lidos
+        );
```

Comment:
[BUG]
Se o `conteudo` for maior que ~`BUFFER_SIZE - 60` de tamanho, Essa linha de código vai falhar e corromper seu programa.